typing/typecore.ml

```diff
@@ -2034,16 +2021,43 @@ and type_pat_aux
                                 loc, sp.ppat_attributes) :: p.pat_extra }
   | Ppat_exception p ->
       let p_exn = type_pat tps Value p Predef.type_exn in
-      rcp {
+      rcp ~exn:true {
         pat_desc = Tpat_exception p_exn;
         pat_loc = sp.ppat_loc;
         pat_extra = [];
         pat_type = expected_ty;
         pat_env = !!penv;
         pat_attributes = sp.ppat_attributes;
       }
-  | Ppat_effect _ ->
-      raise (Error (loc, !!penv, Effect_pattern_below_toplevel))
+  | Ppat_effect (p,k) ->
+      let eff_ty, cont_ty = match pat_expected_ty.eff_typ with
+        | None ->
+            (* effect type are generated in the type_pattern function
+               in presence of effect patterns. *)
+            assert false
+        | Some (e,c) -> e, c
+      in
+      let k =
+        match k.ppat_desc with
+        | Ppat_any -> None
+        | Ppat_var name ->
+            let ty = instance cont_ty in
+            let id, _uid =
```

Comment:
I think it would be useful to store the uid of the continuation in the `Tpat_effect` node in the same way as we do for `Tpat_var` and `Tpat_alias`.  (Merlin rely on this information when querying local occurrences from an identifier's declaration.)